File: Python3/601 - 800/633 Sum of Square Numbers.py
# 272ms, 52.44%; 14.2MB, 50.00%
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for i in range(0, 1 + int((c // 2) ** 0.5)):
            tmp = c - i ** 2
            if tmp == int(tmp ** 0.5) ** 2: return True
        return False

File: 601 - 800/633. Sum of Square Numbers.cc
// 0ms, 100.00%; 8.1MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool judgeSquareSum(int c) {
        for (int i = 0, limit = 1 + sqrt(c / 2); i < limit; ++i) {
            int tmp = c - i * i;
            if (tmp == pow(int(sqrt(tmp)), 2)) return true;
        }
        return false;
    }
};
